src/interfaces/wallet.cpp

```diff
@@ -79,15 +75,19 @@ WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx)
         result.txout_address_is_mine.emplace_back(ExtractDestination(txout.scriptPubKey, result.txout_address.back()) ?
                                                       IsMine(wallet, result.txout_address.back()) :
                                                       ISMINE_NO);
-        // NOTE:  It's ok to check every output here, we know all the output amounts
-        if (result.is_denominate && !CPrivateSend::IsDenominatedAmount(txout.nValue)) {
-            result.is_denominate = false;
+        if (!fOutputDenomFound && result.txout_address_is_mine.back() && !CPrivateSend::IsDenominatedAmount(txout.nValue)) {
+            fOutputDenomFound = true;
         }
     }
+    result.credit = wtx.GetCredit(ISMINE_ALL);
+    result.debit = wtx.GetDebit(ISMINE_ALL);
     result.change = wtx.GetChange();
     result.time = wtx.GetTxTime();
     result.value_map = wtx.mapValue;
     result.is_coinbase = wtx.IsCoinBase();
+    result.is_denominate = wtx.tx->vin.size() == wtx.tx->vout.size() && // Number of inputs is same as number of outputs
+                           (result.credit - result.debit) == 0 && // Transaction pays no tx fee
```

Comment:
nit: imo this is a bit more readable
```suggestion
    result.is_denominate = (wtx.tx->vin.size() == wtx.tx->vout.size()) && // Number of inputs is same as number of outputs
                           (result.credit - result.debit == 0) && // Transaction pays no tx fee
```